disnake/state.py

```diff
@@ -930,6 +931,33 @@ def parse_message_reaction_remove_emoji(
                 if reaction:
                     self.dispatch("reaction_clear_emoji", reaction)
 
+    def _handle_poll_event(
+        self, raw: RawMessagePollVoteActionEvent, event_type: Literal["add", "remove"]
+    ) -> None:
+        guild = self._get_guild(raw.guild_id)
+        answer = None
+        if guild is not None:
+            member = guild.get_member(raw.user_id)
+            message = self._get_message(raw.message_id)
+            if message is not None and message.poll is not None:
+                answer = utils.get(message.poll.answers, id=raw.answer_id)
```

Comment:
Side note, linear search might be expensive if the poll is in a large server with a bunch of activity. Can answers be stored in a dict somehow, maybe?

Reply:
We could store them in a dict which has the keys that are the message ID and the values a list of Answer(s)

Reply:
I was thinking a `{answer_id: PollAnswer}` dict on `Poll` objects.
Trying to keep such a dict in sync with the existing answer list would be error-prone; it might be possible to use said dict internally and have a `@property` that returns the `List[PollAnswer]` instead?

it's fine if this doesn't end up working out, but I'm concerned about the performance hit of `utils.get` here nonetheless.

Reply:
I feel that this is the best thing to do but there's an issue. `Poll` accepts `List[PollAnswer]` to make the user able to construct a poll to be sent with the bot, now how should that work with the property? The IDs are assigned by discord, i think they assign them based on the order of the answers but i'm not sure, and relying on this is dangerous.

Reply:
Ah, fair point. You're absolutely right that relying on ID assignments is dangerous, but that's not what I was going for.
The dict key can be considered an implementation detail, and doesn't have to line up with Discord's IDs initially. User-created `Poll` objects should never end up in this part of the code, and `Poll` objects from the API will always have correct answer IDs.

But yeah, either way, maybe it would be wise to first resolve everything else (especially class hierarchies) before trying to tackle this, having shared classes makes this a bit weird.

Reply:
Commenting this here as a reminder for my future self (as i have alzhaimer probably), as mentioned on discord, this would really benefit us for the sake of combining `PollAnswerCount` into `PollAnswer`, what makes this really complicated is the structure that the API returns, consider this payload from the API:

```json
{
   "type":0,
   "content":"",
   "mentions":[],
   "mentionroles":[],
   "attachments":[],
   "embeds":[],
   "timestamp":"2024-08-07T15:10:08.887000+00:00",
   "editedtimestamp":"None",
   "flags":0,
   "components":[],
   "id":"1270760903066783848",
   "channel_id":"838179278943551518",
   "author":{
      "id":"710570210159099984",
      "username":"__snipy",
      "avatar":"78d8610364e807dd35f0c20cff13dc90",
      "discriminator":"0",
      "public_flags":256,
      "flags":256,
      "banner":"None",
      "accent_color":"None",
      "global_name":"None",
      "avatar_decoration_data":"None",
      "banner_color":"None",
      "clan":"None"
   },
   "pinned":false,
   "mention_everyone":false,
   "tts":false,
   "position":0,
   "poll":{
      "question":{
         "text":"hhhhh"
      },
      "answers":[
         {
            "answer_id":1,
            "poll_media":{
               "text":"hhh"
            }
         },
         {
            "answer_id":2,
            "poll_media":{
               "text":"hhh?!"
            }
         }
      ],
      "expiry":"2024-08-08T15:10:08.883282+00:00",
      "allow_multiselect":false,
      "layout_type":1,
      "results":{
         "answer_counts":[
            {
               "id":1,
               "count":1,
               "me_voted":false
            }
         ],
         "is_finalized":false
      }
   }
}
```
as you can see the data that is holded by `PollAnswerCount` is at `data["poll"]["results"]["answer_counts"]`, we want that data to be folded into `PollAnswer` that is at `data["poll"]["answers"]`